x-element.js

```diff
@@ -1385,7 +1387,14 @@ class TemplateEngine {
             string = string.slice(0, -3 - name.length) + `${TemplateEngine.#ATTRIBUTE_PREFIXES.property}${iii}="${name}`;
             state.index = 1; // Accounts for an expected quote character next.
           } else {
-            throw new Error(`Found invalid template string "${string}" at "${string.slice(state.index)}".`);
+            // We try and intuit what authors consider line “1”.
+            const handled = [...strings.slice(0, iii), string.slice(0, state.index)].join('');
+            const lineCount = handled.startsWith('\n')
+              ? handled.split('\n').length - 1
+              : handled.split('\n').length;
+            // We say “on or after” because interpolated JS could add lines we
+            //  cannot know about or account for.
```

Comment:
Note that you _can_ write templates like this, hence the “on or after” phrase:

```js
html`
  <div .ok="${







                           'i can haz lines'


                                                }" .oops='${whoops}'>
  </div>
`;
```

This is gonna report like “on or after line 1” and there’re really not much we can do about that!